execution/execution_test.go

```diff
@@ -0,0 +1,174 @@
+package execution
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/mesg-foundation/core/service"
+)
+
+var (
+	srv = &service.Service{
+		Tasks: []*service.Task{
+			&service.Task{
+				Key: "task",
+				Inputs: []*service.Parameter{
+					&service.Parameter{Key: "foo", Type: "String"},
+					&service.Parameter{Key: "bar", Type: "String"},
+				},
+				Outputs: []*service.Output{
+					&service.Output{
+						Key: "outputX",
+						Data: []*service.Parameter{
+							&service.Parameter{
+								Key:  "foo",
+								Type: "String",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	taskKey       = "task"
+	defaultInputs = map[string]interface{}{
+		"foo": "hello",
+		"bar": "world",
+	}
+	tags = []string{"tag1", "tag2"}
+)
+
+func db(t *testing.T, dir string) DB {
+	fmt.Println(dir)
+	db, err := New(dir)
+	require.NoError(t, err)
+	return db
+}
+
+func TestCreate(t *testing.T) {
+	db := db(t, filepath.Join(os.TempDir(), "TestCreate"))
+	defer db.Close()
+	tests := []struct {
+		inputs map[string]interface{}
+		assert bool
+	}{
+		{inputs: map[string]interface{}{"foo": "hello", "bar": "world"}, assert: false},
+		{inputs: map[string]interface{}{"foo": "hello"}, assert: true},
+	}
+	for _, test := range tests {
+		execution, err := db.Create(srv, taskKey, test.inputs, tags)
+		if test.assert {
+			require.Error(t, err)
+		} else {
+			require.NoError(t, err)
+			require.NotNil(t, execution)
+			e, err := db.Find(execution.ID)
+			require.NoError(t, err)
+			require.NotNil(t, e)
+			require.Equal(t, srv.ID, e.Service.ID)
+			require.Equal(t, taskKey, e.TaskKey)
+			require.Equal(t, test.inputs, e.Inputs)
+			require.Equal(t, tags, e.Tags)
+			require.Equal(t, execution.Status, Created)
+			require.NotZero(t, e.CreatedAt)
+		}
+	}
+}
+
+func TestFind(t *testing.T) {
+	db := db(t, filepath.Join(os.TempDir(), "TestFindExecution"))
+	defer db.Close()
+	e, _ := db.Create(srv, taskKey, defaultInputs, tags)
+	tests := []struct {
+		id     string
+		assert bool
+	}{
+		{e.ID, false},
+		{"noid", true},
```

Comment:
keep consistency with tests:

change to "dosen't exists" or "noid" for all tests